packages/spacecat-shared-data-access/src/models/api-key/api-key.js

```diff
@@ -43,6 +43,28 @@ const ApiKey = (data) => {
   self.getDeletedAt = () => self.state.deletedAt;
   self.getScopes = () => self.state.scopes;
 
+  /**
+   * Checks if the apiKey is valid.
+   * @returns {boolean} True if the apiKey is valid, false otherwise
+   */
+  self.isValid = () => {
+    const now = new Date();
+
+    if (self.state.deletedAt && new Date(self.state.deletedAt) > now) {
+      return false;
+    }
+
+    if (self.state.revokedAt && new Date(self.state.revokedAt) > now) {
+      return false;
+    }
+
+    if (self.state.expiresAt && new Date(self.state.expiresAt) > now) {
+      return false;
+    }
```

Comment:
Same issue with these two, I think.

If an expiry is set when the key is created (as it must) than it will always be in the future, which would mean that this check fails and the key is not considered valid.